src/layman/upgrade/__init__.py

```diff
@@ -0,0 +1,63 @@
+import logging
+import psycopg2
+from operator import itemgetter
+
+from layman.upgrade import upgrade_v1_8, upgrade_v1_9
+from layman import settings
+from layman.common.prime_db_schema import util as db_util
+DB_SCHEMA = settings.LAYMAN_PRIME_SCHEMA
+
+logger = logging.getLogger(__name__)
+
+
+DATA_MIGRATIONS = [
+    ((1, 9, 0), [upgrade_v1_9.initialize_data_versioning, ]),
+]
+
+
+def get_max_data_version():
+    max_migration = max(DATA_MIGRATIONS, key=itemgetter(0))
+    return max_migration[0] + (len(max_migration[1]) - 1, )
+
+
+def get_current_data_version():
+    # This table should have only one row (or none for the first time)
+    try:
+        sql_select = f'''select max(major_version), max(minor_version), max(patch_version), max(migration), count(*) from {DB_SCHEMA}.data_version;'''
+        current_version_sql = db_util.run_query(sql_select, encapsulate_exception=False)[0]
+        row_count = current_version_sql[4]
+        if row_count == 1:
+            current_version = (current_version_sql[0], current_version_sql[1], current_version_sql[2], current_version_sql[3])
+        elif row_count == 0:
+            current_version = (-1, -1, -1, -1)
+        else:
+            assert row_count == 1
+    except psycopg2.errors.UndefinedTable:
+        current_version = (-1, -1, -1, -1)
+    return current_version
+
+
+def upgrade():
+    logger.info(f'Checking all upgrades')
+    if upgrade_v1_8.older_than_1_8():
+        upgrade_v1_8.upgrade_1_8()
+
+    current_data_version = get_current_data_version()
+    max_data_version = get_max_data_version()
+    logger.info(f'  Current data version = {current_data_version}, Maximum data version = {max_data_version}')
+    migration_list_full = list(filter(lambda x: x[0] > current_data_version, DATA_MIGRATIONS))
```

Comment:
Consider situation of two migrations
```
DATA_MIGRATIONS = [
    ((1, 9, 0), [migration1, migration2]),
]
```
Some instance migrated migration1 sucessfully, but migration2 not. After some manual change sysadmin re-run the migration. As a result, migration2 should be applied.